lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -165,15 +167,12 @@ impl ModuleInternal for DlModule {
         Ok(unsafe { from_raw_parts(*p_table_segment, **p_table_segment_len as usize) })
     }
 
-    fn get_export_func(&self, sym: &[u8]) -> Result<FunctionHandle, Error> {
-        let mut guest_sym: Vec<u8> = b"guest_func_".to_vec();
-        guest_sym.extend_from_slice(sym);
-
+    fn get_export_func(&self, sym: &str) -> Result<FunctionHandle, Error> {
```

Comment:
This became another one of those "oh, I'll start changing more stuff while I'm here!" situations.

@pchickey since we talked earlier and realized `function_id_by_name` is actually too broad for this use, I changed it to `get_export_func_id`, which more closely matches the primary user, `get_export_func`, anyway. Because it goes through export names, it compares functions without the `guest_func_` prefix. Because export names are valid utf8 by [The Spec](https://webassembly.github.io/spec/core/syntax/values.html#syntax-name) this can take `&str` instead of a `&[u8]`. That simplifies things and avoids questions about "what if sym is a byte slice that is not utf8", which should never happen anyway.

Reply:
Ooo, that's good to know! Rust makes bytestrings easy enough to write but I was always annoyed by that type showing up in the region traits